server/src/test/java/io/spine/examples/shareaware/server/e2e/given/E2ETestUser.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.shareaware.server.e2e.given;
+
+import com.google.common.collect.ImmutableList;
+import io.spine.base.CommandMessage;
+import io.spine.base.EntityState;
+import io.spine.base.EventMessage;
+import io.spine.client.Client;
+import io.spine.client.Subscription;
+import io.spine.core.UserId;
+import io.spine.examples.shareaware.WalletId;
+import io.spine.examples.shareaware.server.given.GivenWallet;
+import io.spine.examples.shareaware.wallet.command.CreateWallet;
+import io.spine.testing.core.given.GivenUserId;
+
+import java.util.concurrent.CompletableFuture;
+
+import static io.spine.examples.shareaware.server.given.GivenWallet.*;
+
+/**
+ * Represents a user for end-to-end tests which interacts
+ * with the server-side with the help of {@link Client}.
+ *
+ * <p>In case when the user is needed to have actions that
```

Comment:
```
In some tests, it may be required to extend the API provided by this type. 
Then, subclassing should be done in a nested type, residing inside 
the target test class.
```